### ISSUE TITLE
Problem running copy-dsyms.sh scripts

### ISSUE THREAD

**Issue by vizeke** [CocoaPods/CocoaPods#10488]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

When I build an app that runs XXXX-copy-dsyms.sh scripts, I get this error
```
touch: /Users/mutuallife/Library/Developer/Xcode/DerivedData/App-gqrxxdkopuweewejrpeccjjpxuqi/Build/Products/Debug (dev)-iphoneos/ZendeskChatProvidersSDK/ChatProvidersSDK.framework.ios-i386_x86_64-simulator.dSYM: No such file or directory
Command PhaseScriptExecution failed with a nonzero exit code
```

On this line
```
# The dSYM was not stripped at all, in this case touch a fake folder so the input/output paths from Xcode do not reexecute this script because the file is missing.
touch "${DWARF_DSYM_FOLDER_PATH}/${basename}.dSYM"
```

Which is probably generated by this file
https://github.com/CocoaPods/CocoaPods/blob/3593ac03a661fbdd7479a648062a42065d34412c/lib/cocoapods/generator/script_phase_constants.rb#L82

For now, I need to adjust the scripts manually by adding `mkdir -p "${DWARF_DSYM_FOLDER_PATH}"` prior to the touch command, in each of the copy-dsyms scripts.
I don't know if that is supposed to be a real solution, that is why I didn't make a PR.

## What did you do?

1. Installed a new library `ZendeskChatSDK`
2. Tried to build my project

## What did you expect to happen?

Expected my build to run correctly.

## What happened instead?

I get an error for the package that runs the copy-dsyms script. Caused by not having the complete path prior to a touch command.

![image](https://user-images.githubusercontent.com/1816558/109843027-6aaa9680-7c29-11eb-84cc-8c325f0f0fe5.png)

### Stack
```
   CocoaPods : 1.10.1
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin20]
    RubyGems : 3.0.3
        Host : macOS 11.1 (20C69)
       Xcode : 12.3 (12C33)
         Git : git version 2.25.0
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib
Repositories : cocoapods - git - https://github.com/CocoaPods/Specs.git @ 98d6e41de58506808a3352f36165993166c0caa3
               trunk - CDN - https://cdn.cocoapods.org/
```
### Installation Source
```
Executable Path: /usr/local/bin/pod
```
### Plugins
```
cocoapods-deintegrate : 1.0.4
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.4.1
cocoapods-try         : 1.1.0
```
### Podfile
```ruby
platform :ios, ‘11.0’
use_frameworks!
# workaround to avoid Xcode 10 caching of Pods that requires
# Product -> Clean Build Folder after new Cordova plugins installed
# Requires CocoaPods 1.6 or newer
install! ‘cocoapods’, :disable_input_output_paths => true
def capacitor_pods
  # Automatic Capacitor Pod dependencies, do not delete
  pod ‘Capacitor’, :path => ‘../../../../node_modules/@capacitor/ios’
  pod ‘CapacitorCordova’, :path => ‘../../../../node_modules/@capacitor/ios’
  pod ‘CapacitorCommunityFirebaseCrashlytics’, :path => ‘../../../../node_modules/@capacitor-community/firebase-crashlytics’
  pod ‘TriplesenseCapacitorDeviceInfo’, :path => ‘../../../../node_modules/@triplesense/capacitor-device-info’
  pod ‘TriplesenseCapacitorImageCropx’, :path => ‘../../../../node_modules/@triplesense/capacitor-image-cropx’
  pod ‘CordovaPlugins’, :path => ‘../capacitor-cordova-ios-plugins’
  pod ‘CordovaPluginsStatic’, :path => ‘../capacitor-cordova-ios-plugins’
  pod ‘CordovaPluginsResources’, :path => ‘../capacitor-cordova-ios-plugins’
  # Do not delete
end
def rudder_pods
  pod ‘Rudder’, ‘1.0.7’
  pod ‘Rudder-Firebase’
  pod ‘Rudder-Facebook’, ‘1.0.0’
end
def zendesk_pods
  pod ‘ZendeskAnswerBotSDK’
  pod ‘ZendeskChatSDK’
  pod ‘ZendeskSupportSDK’
end
target ‘App’ do
  capacitor_pods
  # Add your Pods here
  rudder_pods
  zendesk_pods
  pod ‘Branch’
end
```

## Project that demonstrates the issue

https://github.com/zendesk/ios_sdk_demo_apps/blob/master/SupportSDKSamples/README.md

But you need to build without having one of the possible architectures.

**Comment by dnkoutso** [comment]
@vizeke seems like an issue yes, lets update to call `mkdir -p`?

**Comment by dnkoutso** [comment]
fix was merged for 1.11.0
